### DISCUSSION TITLE
When to use `modern_sqlite` and `vtab` features with loadable extensions

### DISCUSSION THREAD

**Discussion by nyurik** [rusqlite/rusqlite#1423] [Questions]
I need to covert into loadable (.so/.dll) my two extensions [sqlite-hashes](https://github.com/nyurik/sqlite-hashes) (md5,sha*,...) and [sqlite-compressions](https://github.com/nyurik/sqlite-compressions) (gzip,brotli), supporting oldest possible SQLite.  I ran into a few, mostly documentation(?) issues:

* Enabling `modern_sqlite` feature automatically makes my extension non-loadable because `Error: error during initialization: SQLite version mismatch:  runtime v3.37.2 < compiled v3.44.0`.  The `modern_sqlite` is used in the `loadable_extension` example - do I need it?
* The `loadable_extension` example uses `vtab` - what is it, and do I need it?  The example seems to not need it, but uses it?
* Enabling Window function support breaks unless `bundled` is feature is enabled. I get `SQLite API not initialized or SQLite feature omitted` error otherwise (from inside `sqlite3_threadsafe()`).  Is there a way I can enable window support without bundling? I would assume bundling is wrong when building a loadable extension?

Thanks!

**Comment by gwenn** [comment]
Indeed, neither `modern_sqlite` nor `vtab` features are needed:
```diff
$ git diff
diff --git a/Cargo.toml b/Cargo.toml
index f8d915e..524e24b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -163,11 +163,11 @@ harness = false
 [[example]]
 name = "loadable_extension"
 crate-type = ["cdylib"]
-required-features = ["loadable_extension", "modern_sqlite", "functions", "vtab", "trace"]
+required-features = ["loadable_extension", "functions", "trace"]

 [[example]]
 name = "load_extension"
-required-features = ["load_extension", "bundled", "functions", "vtab", "trace"]
+required-features = ["load_extension", "bundled", "functions", "trace"]

 [package.metadata.docs.rs]
 features = ["modern-full", "rusqlite-macros"]
diff --git a/examples/loadable_extension.rs b/examples/loadable_extension.rs
index e913240..5a197d3 100644
--- a/examples/loadable_extension.rs
+++ b/examples/loadable_extension.rs
@@ -8,7 +8,7 @@ use rusqlite::{to_sqlite_error, Connection, Result};

 /// # build
 /// ```sh
-/// cargo build --example loadable_extension --features "loadable_extension modern_sqlite functions vtab trace"
+/// cargo build --example loadable_extension --features "loadable_extension functions trace"
 /// ```
 /// # test
 /// ```sh
```

**Comment by gwenn** [comment]
For window function, https://www.sqlite.org/windowfunctions.html#history:
> Window function support was first added to SQLite with release [version 3.25.0](https://www.sqlite.org/releaselog/3_25_0.html) (2018-09-15).

And, by default, rusqlite uses 3.14.0:
https://github.com/rusqlite/rusqlite/blob/master/README.md
> The base rusqlite package supports SQLite version 3.14.0
> min_sqlite_version_3_14_0 - SQLite 3.14.0 bindings (this is the default)

So you need to compile `rusqlite` against your own version (>= 3.2.5.0) of SQLite.
